### PR TITLE
[REVIEW] Specialized hash function for string_view

### LINKED ISSUE #3006: [FEA] Specialize hash function for `string_view`
**Is your feature request related to a problem? Please describe.**

In order to be able to use the new STRING column introduced in https://github.com/rapidsai/cudf/pull/2811 for operations like join/groupby/hash_partition, we need to be able to compute a hash value from a `string_view` element. 

`MurmurHash3_32` needs to be specialized for `string_view` to handle hashing the entire string. 

**Describe the solution you'd like**
[`MurmurHash3_32`](https://github.com/rapidsai/cudf/blob/branch-0.8/cpp/src/hash/hash_functions.cuh#L33) needs to be specialized for `string_view` to handle hashing the entire string.

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #3006 
Adding specialized Murmur3_32 for string_view type.
This will handle hashing the string's bytes when a strings column is used for hash-join, etc.

### CODE REVIEW

**Comment by davidwendt** [comment]
rerun tests

**Comment by davidwendt** [comment]
rerun tests

**Comment by davidwendt** [comment]
This is now dependent on #2971 for the test modules.

File: cpp/src/hash/hash_functions.cuh [hunk 1]
```diff
@@ -189,4 +191,57 @@ struct IdentityHash<cudf::detail::wrapper<T,type_id>>
...
+/**
+* @brief Specialization of MurmurHash3_32 operator for strings.
+*/
+template<>
+__forceinline__ 
+__host__ __device__ hash_value_type MurmurHash3_32<cudf::string_view>::operator()(const cudf::string_view& key) const
+{
+    const int len = (int)key.size_bytes();
+    const uint8_t* data = (const uint8_t*)key.data();
+    const int nblocks = len / 4;
```
Comment (jrhemstad) [inline cpp/src/hash/hash_functions.cuh]: what if `len < 4`? Won't you end up hashing nothing?

Comment (davidwendt) [reply cpp/src/hash/hash_functions.cuh]: No. The `tail` section still executes
https://github.com/rapidsai/cudf/blob/637683d07cc6e867e487ec950c043855dc1c6db6/cpp/src/hash/hash_functions.cuh#L231

Comment (davidwendt) [reply cpp/src/hash/hash_functions.cuh]: Test case includes a single character as well.
https://github.com/rapidsai/cudf/blob/637683d07cc6e867e487ec950c043855dc1c6db6/cpp/tests/strings/hash_string.cu#L49

Comment (jrhemstad) [reply cpp/src/hash/hash_functions.cuh]: Ah, okay. Makes sense.

**Comment by kkraus14** [comment]
rerun tests

**Comment by jrhemstad** [comment]
rerun tests

**Comment by davidwendt** [comment]
rerun tests

**Comment by jrhemstad** [comment]
@davidwendt it looks like this is failing due to the `error_utils.hpp` issue. I think you just need to merge in the latest `branch-0.11`.

**Pull request merged**